common/djangoapps/student/views.py

```diff
@@ -753,15 +753,14 @@ def _allow_donation(course_modes, course_id):
     Args:
         course_modes (dict): Mapping of course ID's to course mode dictionaries.
         course_id (str): The unique identifier for the course.
+        enrollment(CourseEnrollment): The enrollment object in which the user is enrolled
 
     Returns:
         True if the course is allowing donations.
 
     """
     donations_enabled = DonationConfiguration.current().enabled
-    is_verified_mode = CourseMode.has_verified_mode(course_modes[course_id])
-    has_payment_option = CourseMode.has_payment_options(course_id)
```

Comment:
If these class methods aren't being used anywhere else, you can delete them from the `CourseMode` model.